activerecord/lib/active_record/connection_adapters/abstract/query_comment.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+require "active_support/core_ext/module/attribute_accessors_per_thread"
+
+module ActiveRecord
+  module ConnectionAdapters
+    # SQL query comments set at the adapter level
+    module QueryComment
+      mattr_accessor :prepend_comment, default: false
+      def add_query_comment?
+        return false unless ActiveRecord::Base.query_comment_enabled
+        return false if Context.components.empty?
+        true
+      end
+
+      def add_query_comment_to_sql(sql)
+        return sql unless add_query_comment?
+        comment = Context.comment
+        if comment.present? && !sql.include?(comment)
+          sql = if prepend_comment
+            "#{comment} #{sql}"
+          else
+            "#{sql} #{comment}"
+          end
+        end
+        sql
+      end
+
+      # Maintains a user-defined context for all queries and constructs an SQL comment by
+      # calling methods listed in :components
+      #
+      # Additional information can be pushed to the context to be referenced by methods
+      # defined in framework or application initializers.
+      module Context
+        mattr_accessor :components, :lines_to_ignore, instance_accessor: false
```

Comment:
(see comment below on `backtrace_cleaner`)

To replace `lines_to_ignore` with `backtrace_cleaner`

```
class_attribute :backtrace_cleaner, default: ActiveSupport::BacktraceCleaner.new
```

Then in the Railtie we'll want to

```
ActiveSupport.on_load(:active_record) { ConnectionAdapters.backtrace_cleaner = ::Rails.backtrace_cleaner }
```

This is the same thing `LogSubscriber` does so we can mostly copy that.

Reply:
> To replace `lines_to_ignore` with `backtrace_cleaner`

Hey thanks for the tip on `backtrace_cleaner` ! You have lost me on this part tho ^. 

`lines_to_ignore` would be a regex to override the default, so seems suitable to pass to `backtrace_cleaner.add_silencer` if present?

Reply:
Sorry, I'm suggesting that this shouldn't have its own `lines_to_ignore` and should instead by default use the configuration of `::Rails.backtrace_cleaner` (a good reasonable default)

Reply:
Ooh - I think I understand..perhaps... This would allow someone to specify their filtering preference by modifying Rails.backtrace_cleaner, rather than passing regexes around as config options?

Reply:
Exactly!